test/integration/admin/api/api_docs_services_controller_test.rb

```diff
@@ -103,6 +103,83 @@ def test_update_unexistent_service
       assert_response :not_found
     end
 
+    class MemberPermissions < ActionDispatch::IntegrationTest
+      setup do
+        @provider = FactoryBot.create(:simple_provider)
+        @accessible_service = FactoryBot.create(:simple_service, account: provider)
+        @forbidden_service = FactoryBot.create(:simple_service, account: provider)
+        @accessible_api_docs_service = FactoryBot.create(:api_docs_service, account: provider, service: accessible_service)
+        @forbidden_api_docs_service = FactoryBot.create(:api_docs_service, account: provider, service: forbidden_service)
+        @member = FactoryBot.create(:member, account: provider, admin_sections: ['partners'])
+        @access_token = FactoryBot.create(:access_token, owner: member, scopes: %w[account_management], permission: 'rw')
+
+        member.member_permission_service_ids = [accessible_service.id]
+        member.save!
+
+        host! provider.admin_domain
+      end
+
+      attr_reader :provider, :accessible_service, :forbidden_service, :accessible_api_docs_service, :forbidden_api_docs_service, :member, :access_token
+
+      test 'index' do
+        get admin_api_active_docs_path(path_params)
+        assert_response :success
+        api_docs_services_ids = JSON.parse(response.body)['api_docs'].map { |api_doc| api_doc.dig('api_doc', 'id') }
+        assert_equal [accessible_api_docs_service.id], api_docs_services_ids
+      end
+
+      test 'read accessible api doc service' do
+        get admin_api_active_doc_path(accessible_api_docs_service, **path_params)
+        assert_response :success
+      end
+
+      test 'read forbidden api doc service' do
+        get admin_api_active_doc_path(forbidden_api_docs_service, **path_params)
+        assert_response :not_found
+      end
+
+      test 'create with forbidden service' do
+        post admin_api_active_docs_path(path_params), api_doc_params(service_id: forbidden_service.id)
+        assert_response :unprocessable_entity
+      end
+
+      test 'update to forbidden service' do
+        put admin_api_active_doc_path(accessible_api_docs_service, **path_params), { service_id: forbidden_service.id }
+        assert_response :unprocessable_entity
+      end
+
+      test 'read account level service' do
+        account_level_api_docs_service = FactoryBot.create(:api_docs_service, account: provider, service: nil)
+
+        get admin_api_active_doc_path(account_level_api_docs_service, **path_params)
+        assert_response :success
+
+        get admin_api_active_docs_path(path_params)
+        assert_response :success
+        api_docs_services_ids = JSON.parse(response.body)['api_docs'].map { |api_doc| api_doc.dig('api_doc', 'id') }
+        assert_contains api_docs_services_ids, account_level_api_docs_service.id
+      end
+
+      protected
+
+      def path_params
+        { access_token: access_token.value, format: :json }
+      end
+
+      def api_doc_params(**extra_params)
```

Comment:
If this class inherits from `Admin::Api::ApiDocsServicesControllerTest`, you can reuse this method.

We should also remove the `login!` in the parent class because using `login!` for API tests is not a very good idea 😄